src/recommend.rs

```diff
@@ -0,0 +1,76 @@
+use std::ops::Index;
+
+use reqwest::Url;
+use serde_derive::{Deserialize, Serialize};
+
+type ConnectivityResponse = Vec<Nodes>;
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Nodes {
+    pub public_key: String,
+    pub channels: i64,
+    pub capacity: i64,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Node {
+    #[serde(rename = "public_key")]
+    pub public_key: String,
+    pub alias: String,
+    pub sockets: String,
+    #[serde(rename = "active_channel_count")]
+    pub active_channel_count: i64,
+    pub capacity: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Recommendations {
+    pub routing_node: Node,
+    pub edge_node: Node,
+}
+
+pub async fn get_recommended_channels() -> Result<Recommendations, RecommendedError> {
+    let base_url = "https://mempool.space/api/v1/lightning/nodes/rankings/connectivity".to_string();
+    let url = Url::parse(&base_url).map_err(InternalRecommendedError::Url)?;
+
+    let res = reqwest::Client::new().get(url).send().await?;
+    let high_channel_nodes = res.json::<ConnectivityResponse>().await?;
+    let mut high_cap_nodes = high_channel_nodes.clone();
+    high_cap_nodes.sort_by(|a, b| a.capacity.partial_cmp(&b.capacity).unwrap());
+
+    let routing_node = get_node(&high_cap_nodes.index(0).public_key).await?;
+    let edge_node = get_node(&high_channel_nodes.index(0).public_key).await?;
+
+    Ok(Recommendations { routing_node, edge_node })
+}
+
+async fn get_node(pubkey: &str) -> Result<Node, RecommendedError> {
+    let base_url = format!("https://mempool.space/api/v1/lightning/nodes/{}", pubkey);
+    let url = Url::parse(&base_url).map_err(InternalRecommendedError::Url)?;
+
+    let res = reqwest::Client::new().get(url).send().await?;
+    let node: Node = res.json::<Node>().await?;
+
+    Ok(node)
+}
```

Comment:
Can we use Rust's strong typing abilities to improve this interface?

Reply:
Im not sure what the ask is?

Reply:
Why does this interface take `pubkey: &str`? A better choice might be a strong type that passed validation. e.g. [`ln_types::P2PAddress`](https://docs.rs/ln-types/0.1.4/ln_types/p2p_address/struct.P2PAddress.html)

Reply:
Almost every field in this file would benefit from strong types. No reason to pass capacity, pubkeys, sockets, etc. as `String` types